.github/workflows/precommit.yml

```diff
@@ -30,6 +30,7 @@ jobs:
           - { name: 'Linux LLVM+libstdc++', os: 'ubuntu-22.04', cc: clang, cxx: clang++, config: ''}
           - { name: 'Linux LLVM+libc++', os: 'ubuntu-22.04', cc: clang, cxx: clang++, config: --config=libc++ }
           - { name: 'MacOS LLVM+libc++', os: 'macos-12', cc: clang, cxx: clang++, config: --config=libc++ }
+          - { name: 'M1 MacOS LLVM+libc++', os: 'macos-latest', cc: clang, cxx: clang++, config: --config=libc++ }
```

Comment:
> Wait, `macos-lastest` is the only way to request an ARM machine? That seems awfully brittle. What if latest starts meaning something else? And how do we retain the x86_64 versions?
> 
> 
> 
> Could you provide some refefences for how the actual hardware is selected for the action runners? Then let's try to find a robust and stable way to get that configuration.

Here's the doc that motivated me to work on it: https://github.com/github/roadmap/issues/528

Reply:
I haven't tested them out personally yet, soon I'll create a dummy repository to test out all the various options with focus on specific version instead of the `-latest` tag to avoid the brittle nature.

Reply:
You can just use this pull request to run the actions if you like...

Reply:
Hi @tkoeppe, so I tried the GitHub actions thing and there's a catch. The Github Actions hosted runner is in **public beta** so not everyone got access to it I assume. A completely public option is to use a self-hosted runner deployed on our own M1 machine, do we have any available M1 macs that could do the job?

Also, as an alternative, I can volunteer to build the project on my mac and prepare the `pip` file for it which can be distributed easily. Would that be okay?